applications/FluidDynamicsApplication/custom_processes/embedded_skin_visualization_process.cpp

```diff
@@ -0,0 +1,590 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/checks.h"
+#include "includes/model_part.h"
+#include "utilities/variable_utils.h"
+#include "utilities/divide_triangle_2d_3.h"
+#include "utilities/divide_tetrahedra_3d_4.h"
+#include "modified_shape_functions/triangle_2d_3_modified_shape_functions.h"
+#include "modified_shape_functions/tetrahedra_3d_4_modified_shape_functions.h"
+#include "modified_shape_functions/triangle_2d_3_ausas_modified_shape_functions.h"
+#include "modified_shape_functions/tetrahedra_3d_4_ausas_modified_shape_functions.h"
+
+// Application includes
+#include "embedded_skin_visualization_process.h"
+#include "fluid_dynamics_application_variables.h"
+
+namespace Kratos
+{
+
+/* Public functions *******************************************************/
+EmbeddedSkinVisualizationProcess::EmbeddedSkinVisualizationProcess(
+    ModelPart& rModelPart,
+    ModelPart& rVisualizationModelPart,
+    const std::vector<Variable< double> > VisualizationScalarVariables,
+    const std::vector<Variable< array_1d<double, 3> > > VisualizationVectorVariables,
+    const std::vector<VariableComponent<VectorComponentAdaptor< array_1d< double, 3> > > > VisualizationComponentVariables,
+    const std::string ShapeFunctions,
+    const bool ReformModelPartAtEachTimeStep) : 
+    Process(), 
+    mrModelPart(rModelPart), 
+    mrVisualizationModelPart(rVisualizationModelPart), 
+    mVisualizationScalarVariables(VisualizationScalarVariables),
+    mVisualizationVectorVariables(VisualizationVectorVariables),
+    mVisualizationComponentVariables(VisualizationComponentVariables),
+    mShapeFunctions(ShapeFunctions),
+    mReformModelPartAtEachTimeStep(ReformModelPartAtEachTimeStep){ 
+}
+
+EmbeddedSkinVisualizationProcess::EmbeddedSkinVisualizationProcess(
+    ModelPart& rModelPart,
+    ModelPart& rVisualizationModelPart,
+    Parameters& rParameters) : 
+    Process(), 
+    mrModelPart(rModelPart), 
+    mrVisualizationModelPart(rVisualizationModelPart) {
+
+    Parameters default_parameters( R"(
+    {
+        "shape_functions"                     : "standard",
+        "reform_model_part_at_each_time_step" : false,
+        "visualization_variables"             : ["VELOCITY","PRESSURE","DISTANCE"]
+    })");
+
+    rParameters.ValidateAndAssignDefaults(default_parameters);
+
+    mShapeFunctions = rParameters["shape_functions"].GetString();
```

Comment:
you should check if the shape_functions name is in the list of admissible options